java-checks/src/main/java/org/sonar/java/checks/naming/BadTestClassNameCheck.java

```diff
@@ -38,7 +38,9 @@
 @Rule(key = "S3577")
 public class BadTestClassNameCheck extends IssuableSubscriptionVisitor {
 
-  private static final String DEFAULT_FORMAT = "^((Test|IT)[a-zA-Z0-9]+|[A-Z][a-zA-Z0-9]*(Test|Tests|TestCase|IT|ITCase))$";
+  private static final String STARTING_WITH_TEST = "(Test|IT)[a-zA-Z0-9_]+";
```

Comment:
It looks like you can use '\w' in regex which means exactly the same : 
[a-zA-Z_0-9 (https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html)

On the other hand, it might not be that obvious for others, so up to you.

Reply:
In fact, I can not really easily change this, or it will change the message being reported as well. So I'll let it like this, as I don't feel displaying `\w` in the issue message is nice, and replacing it before printing looks a bit too complex for what it is.